File: solutions/2418. Sort the People/solution.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<string> res;
        std::map<int, string> m;
        for(int i = 0; i < heights.size(); ++i){
            m[heights[i]] = names[i];
        }
        for (auto i = m.rbegin(); i != m.rend(); i++){
            res.push_back(i->second);
        }
        return res;
    }
};

File: solutions/2418. Sort the People/solution.py
class Solution(object):
    def sortPeople(self, names, heights):
        """
        :type names: List[str]
        :type heights: List[int]
        :rtype: List[str]
        """
        d = {}
        for n,h in zip(names,heights):
            d[h] = n
        dKeys = list(d.keys())
        dKeys.sort(reverse=True)
        return [d[k] for k in dKeys]
